.github/workflows/build.yml

```
@@ -101,6 +101,122 @@ jobs:
+        run: |
+          Invoke-WebRequest -Uri https://storage.googleapis.com/chrome-infra/depot_tools.zip -OutFile depot_tools.zip
+          7z x -y -o"C:\workspace\depot_tools" .\depot_tools.zip
+          echo "C:\workspace\depot_tools" >> $Env:GITHUB_PATH
+
+      - name: GClient Sync
+        working-directory: C:\workspace\engine
+        run: |
+          gclient config --name=src\flutter --unmanaged https://github.com/flutter-tizen/engine
+          $env:PYTHONPATH="C:\workspace\depot_tools"
+          python3 src\flutter\ci\tizen\gclient-shallow-sync.py src\flutter\DEPS
+          gclient sync -v --no-history --shallow
+
+      - name: Build
+        working-directory: C:\workspace\engine\src
+        run: |
+          python3 .\flutter\tools\gn --target-os=linux --linux-cpu=${{ matrix.arch }} --no-goma --runtime-mode=${{ matrix.mode }}
+          ninja -C .\out\linux_${{ matrix.mode }}_${{ matrix.arch }} gen_snapshot
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: tizen-${{ matrix.arch }}-${{ matrix.mode }}_windows-x64
+          path: |
+            C:\workspace\engine\src\out\linux_${{ matrix.mode }}_${{ matrix.arch }}\gen_snapshot.exe
+
+  macos-build:
+    runs-on: macos-11
+
+    strategy:
+      matrix:
+        arch: [arm, arm64]
+        mode: [release, profile]
+
+    env:
+      OUTPUT_NAME: linux_${{ matrix.mode }}_${{ matrix.arch }}
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          path: src/flutter
+
+      - uses: actions/cache@v2
+        with:
+          path: src/out/${{ env.OUTPUT_NAME }}
+          key: out-macos-build-${{ env.OUTPUT_NAME }}-${{ github.sha }}
+          restore-keys: |
+            out-macos-build-${{ env.OUTPUT_NAME }}-
+
+      - name: Install depot_tools
+        run: |
+          git clone https://chromium.googlesource.com/chromium/tools/depot_tools.git
+          echo "$PWD/depot_tools" >> $GITHUB_PATH
+      
+      - name: Install required packages
+        run: |
+          brew install md5sha1sum
+
+      - name: GClient Sync
+        run: |
+          gclient config --name=src/flutter --unmanaged https://github.com/flutter-tizen/engine
+          gclient setdep --var=download_android_deps=False --deps-file=src/flutter/DEPS
+          PYTHONPATH="$PWD/depot_tools" python3 ./src/flutter/ci/tizen/gclient-shallow-sync.py src/flutter/DEPS
+          gclient sync -v --no-history --shallow
+      
+      - name: Build
+        run: |
+          src/flutter/ci/tizen/cache-checksum.sh restore src/out/$OUTPUT_NAME
+          # Change host_toolchain to mac/clang_x64.
+          sed -i "" "s|//build/toolchain/linux:clang_$host_cpu|//build/toolchain/mac:clang_$host_cpu|g" src/build/config/BUILDCONFIG.gn
+          # Pass dummy toolchain values to use custom toolchain settings.
+          python3 ./src/flutter/tools/gn --linux --linux-cpu=${{ matrix.arch }} --runtime-mode=${{ matrix.mode }} \
+            --no-goma --macos-build-tizen \
```

For consistency
```suggestion
          python3 ./src/flutter/tools/gn \
            --linux --linux-cpu=${{ matrix.arch }} \
            --runtime-mode=${{ matrix.mode }} \
            --no-goma \
```
By the way haven't we agreed to remove the `--macos-build-tizen` option?

---

I will remove `--macos-build-tizen` parameter and fix the `gn` to use the required settings.